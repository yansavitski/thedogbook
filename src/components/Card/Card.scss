@import 'src/global';

.Card {
  border-radius: $borderRadius;
  box-shadow: 0 0 2px $shadowColor;
  height: 100%;
  min-height: 200px;
  overflow: hidden;
  width: 100%;

  @include transition(transform, 150ms);

  &:hover {
    box-shadow: 0 0 7px 5px $shadowColor;
    transform: scale(1.1);
  }

  &-image {
    height: 100%;
    object-fit: cover;
    width: 100%;
  }
}
