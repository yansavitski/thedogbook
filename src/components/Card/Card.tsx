import './Card.scss'

import React from 'react'

interface CardProps {
  url: string
}

export const Card: React.FunctionComponent<CardProps> = ({
  url,
}: CardProps) => {
  return (
    <figure className="Card">
      <img className="Card-image" src={url} alt="card" loading="lazy" />
    </figure>
  )
}
