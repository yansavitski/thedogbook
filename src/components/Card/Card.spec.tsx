import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'
import React from 'react'

import { Card } from './Card'

describe('<Card />', () => {
  it('renders correctly', () => {
    expect.assertions(1)
    const tree = shallow(<Card url="url to card" />)

    expect(toJson(tree)).toMatchSnapshot()
  })
})
