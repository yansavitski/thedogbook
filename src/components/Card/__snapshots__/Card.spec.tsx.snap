// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> renders correctly 1`] = `
<figure
  className="Card"
>
  <img
    alt="card"
    className="Card-image"
    loading="lazy"
    src="url to card"
  />
</figure>
`;
